components/core/src/clp/BufferedFileReader.hpp

```diff
@@ -1,37 +1,35 @@
 #ifndef CLP_BUFFEREDFILEREADER_HPP
 #define CLP_BUFFEREDFILEREADER_HPP
 
-#include <cstdio>
+#include <cstddef>
 #include <memory>
 #include <optional>
 #include <string>
+#include <utility>
 #include <vector>
 
 #include "BufferReader.hpp"
-#include "Defs.h"
 #include "ErrorCode.hpp"
 #include "ReaderInterface.hpp"
 #include "TraceableException.hpp"
 
 namespace clp {
 /**
- * Class for performing buffered (in memory) reads from an on-disk file with control over when and
- * how much data is buffered. This allows us to support use cases where we want to perform unordered
- * reads from files which only support sequential access (e.g. files from block storage like S3).
+ * Class for performing buffered (in memory) reads from another ReaderInterface with control over
+ * when and how much data is buffered. This allows us to support use cases where we want to perform
+ * unordered reads from input which only support sequential access (e.g. files from block storage
+ * like S3).
  *
  * To control how much data is buffered, we allow callers to set a checkpoint such that all reads
  * and seeks past the checkpoint will be buffered until the checkpoint is cleared. This allows
  * callers to perform random seeks and reads of any data after (and including) the checkpoint.
  * When no checkpoint is set, we maintain a fixed-size buffer.
  *
  * NOTE 1: Unless otherwise noted, the "file position" mentioned in docstrings is the position in
- * the buffered file, not the position in the on-disk file.
+ * the buffered file, not the position in the original input file.
  *
  * NOTE 2: This class restricts the buffer size to a multiple of the page size and we avoid reading
  * anything less than a page to avoid multiple page faults.
- *
- * NOTE 3: Although the FILE stream interface provided by glibc also performs buffered reads, it
```

Comment:
I think this note is still useful to provide context about why we need our own class, right?